Plugin/NonBoltOrderPlugin.php

```diff
@@ -149,21 +146,19 @@ public function execute(Observer $observer)
             $response = $result->getResponse();
             if (empty($response)) {
                 $this->metricsClient->processCountMetric("non_bolt_order_creation.failure", 1);
-                return;
+                return $order;
             }
 
-            $boltTransactionData = ['bolt_transaction_reference' => $response->reference];
-            $payment->setAdditionalInformation(
-                array_merge((array)$payment->getAdditionalInformation(), $boltTransactionData)
-            );
+            $payment->setBoltTransactionReference($response->reference);
```

Comment:
We think that it's possible that `$payment` is null (see line 122). It's not a big deal because we are in `try {}` statement and maybe in real cases, `$payment` is always exists, but probably it's worth fixing it.
We can don't save if we have no payment or save in order table rather than in payment table